### PR TITLE
fix: u64->i64->u64 conversion; chain split height as u64

### PR DESCRIPTION
**Pull request by Impala123**
Description

removing an unnecessary conversion and using u64 for chain height 
---

How Has This Been Tested?
---

rust build & test

### CODE REVIEW

File: applications/tari_base_node/src/grpc/base_node_grpc_server.rs [hunk 1]
```diff
@@ -326,15 +326,15 @@ impl tari_rpc::base_node_server::BaseNode for BaseNodeGrpcServer {
 
         let headers: Vec<u64> = if request.from_height != 0 {
             match sorting {
-                Sorting::Desc => ((cmp::max(0, request.from_height as i64 - num_headers as i64 + 1) as u64)..=
+                Sorting::Desc => (cmp::max(0, request.from_height - num_headers + 1)..=
```
Comment (stringhandler) [inline applications/tari_base_node/src/grpc/base_node_grpc_server.rs]: Is there not a chance for an underflow here?

Comment (delta1) [reply applications/tari_base_node/src/grpc/base_node_grpc_server.rs]: This needs a covering test, even if this logic is moved into a function.

**Comment by delta1** [comment]
Seems you need to run cargo fmt

File: applications/tari_base_node/src/grpc/base_node_grpc_server.rs (hunk 1 above)
Comment (Impala123) [reply applications/tari_base_node/src/grpc/base_node_grpc_server.rs]: should this then be  

`let headers: Vec<u64> = if request.from_height >= num_headers {`

num_headers is forced to be between 10 and 10_000
Means that from_height being 0 would go to the else

Or is it more desirable to check the underflow flag with overflowing_sub() ?

**Review by stringhandler — APPROVED** [review]

**Pull request merged**